doc/designs/disable-stale-users.md

```diff
@@ -0,0 +1,147 @@
+accounts.
+
+Accounts are locked out by setting the nsAccountLock attribute
+(OID: 2.16.840.1.113730.3.1.610) to True.
+
+On very busy clusters, using a non-busy replica (like a hidden replica) for DSU
+is recommended.
+
+
+## Limitations
+
+If regulations call for stale/inactive accounts to be disabled within X
+number of days, then the password lifetime of the passwords for these
+accounts must be set to a value lower than X, minus the expected grace period
+mentioned above, minus the maximum amount of time between two invocations
+of the DSU tool.
+
+
+## Example
+
+Given the following requirements and facts:
+* stale accounts must be disabled after 90 days' inactivity
+* grace period is decided to be 8 days
+* the tool runs every day
+
+setting the maximum password lifetime to 80 days or less would meet the
+guidelines as 80+8+1 is less than 90.
+
+## Multiple runners
+
+DSU can be installed on more than a single replica. This provides redundancy.
+However, care must be taken that DSU is not triggered both:
+* at the same time
+* on the same accounts
+
+So a proper multi-DSU setup has to change both the time of the systemd timer
+the tool is run and the grace period.
+For instance:
+* having one DSU instance start working at 2AM and the other at 4AM
+* making sure the work of the first instance plus the associated replication
+  traffic completes in much less than 2 hours
+* having the first scheduled DSU instance configured to disable accounts with
+  a 7-day grace period and the second instance configured with a 8-day grace
+  period
+
+would make sure that the second DSU instances never disables any account
+unless the first one is offline.
+
+
+## How to Use DSU
+
+### Installation
+
+The Disable Stale Users tooling is contained in the freeipa-server-dsu
+package. 
+``dnf install freeipa-server-dsu`` will install that package.
+
+
+### Configuration
+
+The configuration file for DSU contains the following variables:
+
+```
+[dsu]
+active = False
+grace_period = 7d
+ignore_groups = admins
+log = True
+```
+
+Switch active to True for the DSU tool to disable (lock) accounts.
+
+The grace_period variable is used to set the number of days (or hours) after
+password expiry date on which to disable accounts.
+
+The ignore_groups variable expects a comma separated list of groups. Users
+belonging directly or indirectly to these groups will not be affected.
+
+Log is True by default. The tool will log actions to systemd.
+
+
+### Usage
+
+The DSU tool is not a UNIX daemon. It is a CLI tool that can be triggered
+by a systemd timer. 
+
+Running ``ipa-dsu`` is enough to start the tool. If active is set to True in
+the above-mentioned configuration file, it will disable stale accounts
+appopriately.
+
+The dry-run mode ``ipa-dsu --dry-run`` does not disable/lock accounts in any
+way. Rather, it shows what would be done, essentially acting as a verbose mode
+with "active" set to False. The output is JSON.
```

Comment:
Would a JSON mock-up be a good idea or is that going to be done in the future?

Reply:
I'd rather do it after the fact. I'm undecided on how it's going to look like exactly.